lldb/source/Plugins/Language/Swift/SwiftUnsafeTypes.cpp

```diff
@@ -74,50 +118,120 @@ bool SwiftUnsafeBufferPointer::Update() {
     return false;
 
-  CompilerType argument_type;
+  m_start_addr = start_addr;
 
-  if (CompilerType type = some_value_sp->GetCompilerType())
-    argument_type = SwiftASTContext::GetGenericArgumentType(type, 0);
+  return true;
+}
 
-  if (!argument_type.IsValid())
-    return false;
+class SwiftUnsafeRawBufferPointer final : public SwiftUnsafeType {
+public:
+  SwiftUnsafeRawBufferPointer(ValueObject &valobj);
+  bool Update() override;
 
-  m_elem_type = argument_type;
+private:
+  addr_t m_end_addr;
+};
 
-  ValueObjectSP pointer_value_sp = some_value_sp->GetChildAtIndex(0, true);
-  if (!pointer_value_sp)
+SwiftUnsafeRawBufferPointer::SwiftUnsafeRawBufferPointer(ValueObject &valobj)
+    : SwiftUnsafeType(valobj) {}
+
+bool SwiftUnsafeRawBufferPointer::Update() {
+  if (!m_valobj.GetNumChildren())
     return false;
 
-  addr_t addr = pointer_value_sp->GetValueAsUnsigned(LLDB_INVALID_ADDRESS);
+  // Here is the layout of Swift's UnsafeRaw[Mutable]BufferPointer.
+  // It's a view of the raw bytes of the pointee object. Each byte is viewed as
+  // a `UInt8` value independent of the type of values held in that memory.
+  //
+  // ▿ UnsafeRawBufferPointer
+  //   ▿ _position : Optional<UnsafeRawPointer<Int>>
+  //     ▿ some : UnsafeRawPointer<Int>
+  //       - pointerValue : Int
+  //   ▿ _end : Optional<UnsafeRawPointer<Int>>
+  //     ▿ some : UnsafeRawPointer<Int>
+  //       - pointerValue : Int
+  //
+  // The structure has 2 Optional UnsafePointer to the buffer's start address
+  // and end address. To access the pointer address, lldb unfolds every value
+  // object child until reaching `pointerValue`.
 
+  addr_t addr = GetAddress("_position");
   if (!addr || addr == LLDB_INVALID_ADDRESS)
     return false;
-
   m_start_addr = addr;
 
+  addr = GetAddress("_end");
+  if (!addr || addr == LLDB_INVALID_ADDRESS)
+    return false;
+  m_end_addr = addr;
+
+  if (!m_elem_type.IsValid()) {
+    CompilerType type = m_valobj.GetCompilerType();
+    if (!type.IsValid())
+      return false;
+
+    auto type_system = llvm::dyn_cast<TypeSystemSwift>(type.GetTypeSystem());
+    if (!type_system)
+      return false;
+
+    CompilerType compiler_type =
+        type_system->GetTypeFromMangledTypename(ConstString("$ss5UInt8VD"));
+    if (!compiler_type.IsValid())
+      return false;
+
+    m_elem_type = compiler_type;
+  }
+
+  auto opt_type_size = m_elem_type.GetByteSize(m_valobj.GetTargetSP().get());
+
+  if (!opt_type_size)
+    return false;
+
+  m_count = (m_end_addr - m_start_addr) / *opt_type_size;
+
   return true;
 }
 
+std::unique_ptr<SwiftUnsafeType> SwiftUnsafeType::Create(ValueObject &valobj) {
+  CompilerType type = valobj.GetCompilerType();
+  if (!type.IsValid())
+    return nullptr;
+
+  auto type_system = llvm::dyn_cast<TypeSystemSwift>(type.GetTypeSystem());
+  if (!type_system)
+    return nullptr;
+
+  opaque_compiler_type_t qual_type = type.GetOpaqueQualType();
+
+  if (type_system->IsTypedefType(qual_type))
```

Comment:
GetTypedefType only undoes one level of typedef'ing.  So the result might still be a typedef.  You probably need to loop over IsTypedefType till you get to the non-typedef'ed version.